raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -649,81 +652,172 @@ function receiveTransferExpired(
-
-    const { partnerBalance, partnerLocked, partnerDeposit } = channelAmounts(channel);
-    // don't consider already withdrawn nor pending withdraw requests, since it's a total
-    const totalWithdrawable = partnerDeposit.add(partnerBalance).sub(partnerLocked);
-    assert(
-      request.total_withdraw.lte(totalWithdrawable),
-      'invalid total_withdraw, greater than partner.deposit + own.transferredAmount',
-    );
-
-    if (request.expiration.gt(state.blockNumber)) {
-      // expired request isn't fatal, we still accept the request state change, but don't sign
-      // WithdrawConfirmation and wait for it to expire
-      const confirmation: WithdrawConfirmation = {
-        type: MessageType.WITHDRAW_CONFIRMATION,
-        message_identifier: request.message_identifier,
-        chain_id: request.chain_id,
-        token_network_address: request.token_network_address,
-        channel_identifier: request.channel_identifier,
-        participant: request.participant,
-        total_withdraw: request.total_withdraw,
+/**
+ * Handles a withdraw.request and send a WithdrawRequest to partner
+ *
+ * @param state$ - Observable of current state
+ * @param action - Withdraw request which caused this handling
+ * @param deps - RaidenEpicDeps members
+ * @param deps.address - Our address
+ * @param deps.log - Logger instance
+ * @param deps.signer - Signer instance
+ * @param deps.network - Current Network
+ * @param deps.config$ - Config observable
+ * @returns Observable of withdrawMessage.request|withdraw.failure actions
+ */
+function sendWithdrawRequest(
+  state$: Observable<RaidenState>,
+  action: withdraw.request,
+  { log, address, signer, network, config$ }: RaidenEpicDeps,
+): Observable<withdrawMessage.request | withdraw.failure> {
+  if (action.meta.direction !== Direction.SENT) return EMPTY;
+  return combineLatest([state$, config$]).pipe(
+    first(),
+    mergeMap(([state, { revealTimeout }]) => {
+      const channel = state.channels[channelKey(action.meta)];
+      assert(channel?.state === ChannelState.open, 'channel not open');
+      if (
+        channel.own.pendingWithdraws.some(
+          (req) =>
+            req.total_withdraw.eq(action.meta.totalWithdraw) &&
+            req.expiration.eq(action.meta.expiration),
+        )
+      )
+        return EMPTY; // already requested, skip but don't fail
+      assert(
+        action.meta.expiration >= state.blockNumber + revealTimeout,
+        'expired or expires too soon',
+      );
+      assert(
+        action.meta.totalWithdraw.gt(channel.own.withdraw),
+        'withdraw lower than previous totalWithdraw',
+      );
+      assert(
+        action.meta.totalWithdraw.lte(channelAmounts(channel).ownTotalWithdrawable),
+        'not enough capacity',
+      );
+      const request: WithdrawRequest = {
+        type: MessageType.WITHDRAW_REQUEST,
+        message_identifier: makeMessageId(),
+        chain_id: bigNumberify(network.chainId) as UInt<32>,
```

Comment:
Not much, just a small overhead since `decode` also validates value as `UInt<32>` and would throw if not, while `bigNumberify` just converts to BigNumber and only throws if value isn't conversible to it, since we **know** (that's important when type-asserting with `as`) that this value is in the valid range. So, with this, since it's a value controlled by us, we just tell the compiler to trust us and that this value will always be valid, no need to validate at runtime.
But you're right, `decode(UInt(32), network.chainId)` could be used as well here.